tests/unit/test_file_session_init_store.py

```diff
@@ -0,0 +1,82 @@
+import json
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from openhands.core.config.app_config import AppConfig
+from openhands.server.session.session_init_data import SessionInitData
+from openhands.storage.file_session_init_store import FileSessionInitStore
+from openhands.storage.files import FileStore
+
+
+@pytest.fixture
+def mock_file_store():
+    return MagicMock(spec=FileStore)
+
+
+@pytest.fixture
+def session_init_store(mock_file_store):
+    return FileSessionInitStore(mock_file_store)
+
+
+def test_load_nonexistent_data(session_init_store):
+    session_init_store.file_store.read.side_effect = FileNotFoundError()
+    assert session_init_store.load() is None
+
+
+def test_store_and_load_data(session_init_store):
+    # Test data
+    init_data = SessionInitData(
+        language='python',
+        agent='test-agent',
+        max_iterations=100,
+        security_analyzer='default',
+        confirmation_mode=True,
+        llm_model='test-model',
+        llm_api_key='test-key',
+        llm_base_url='https://test.com',
+        github_token='test-token',
+        selected_repository='test/repo',
```

Comment:
definitely want to remove these two--this config is going to be reused across sessions

IMO the stuff in here should be 1:1 with the stuff in the settings modal